pkg/network/utils/common.go

```diff
@@ -0,0 +1,79 @@
+package utils
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/vishvananda/netlink"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+func getGwAndHostAddressesFromCIDR(s string) (string, string, error) {
+	ip, ipnet, err := net.ParseCIDR(s)
+	if err != nil {
+		return "", "", err
+	}
+
+	subnet, _ := ipnet.Mask.Size()
+	var ips []string
+	for ip := ip.Mask(ipnet.Mask); ipnet.Contains(ip); inc(ip) {
+		ips = append(ips, fmt.Sprintf("%s/%d", ip.String(), subnet))
+
+		if len(ips) == 4 {
+			// remove network address and broadcast address
+			return ips[1], ips[2], nil
+		}
+	}
+
+	return "", "", fmt.Errorf("less than 4 addresses on network")
+}
+
+func inc(ip net.IP) {
+	for j := len(ip) - 1; j >= 0; j-- {
+		ip[j]++
+		if ip[j] > 0 {
+			break
+		}
+	}
+}
+
+func GenerateGatewayAndVmIPAddrs(vmiSpecNetwork *v1.Network, protocol iptables.Protocol) (*netlink.Addr, *netlink.Addr, error) {
```

Comment:
I changed the name of the method since I saw there was existing [bug](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/infraconfigurators/masquerade.go#L138) in the code because of this method confusing name.
But the purpose of the change in this PR is just to move the existing method from driver to here. No to improve the method.